improvement: bigquery specific for query_table_metrics

dongchris opened the pull request:
Using information schema to get row count is much more performant than doing a full table scan

https://cloud.google.com/bigquery/docs/information-schema-table-storage#schema

haritamar commented on macros/edr/materializations/model/metrics.sql:
```diff
@@ -22,6 +22,30 @@
+  {% set metrics = [] %}
+  {% for metric_column in elementary.run_query(query).columns %}
+    {% set metric_name = metric_column.name %}
+    {% set metric_value = metric_column[0] %}
```
Can you add a safety that the query actually returns any results? (it should, but mainly to avoid edge cases / never fail here)

dongchris replied:
can you clarify what is the expected behavior? if `elementary.run_query(query).columns` returns 0 columns, then it just wouldn't update `metrics` dict right?

dongchris replied:
actually, added! @haritamar

haritamar commented:
Thanks @dongchris for the awesome contribution!
Actually realized as well just recently that `count(*)` does a full scan in BQ, we were sure from some reason it's optimized. 

I've added a couple of comments, will be great if you can take a look.
Also worth verifying the the [integration tests](https://github.com/elementary-data/dbt-data-reliability/blob/master/CONTRIBUTING.md#running-integration-tests) pass, but will also check that.

dongchris commented:
> Thanks @dongchris for the awesome contribution! Actually realized as well just recently that `count(*)` does a full scan in BQ and didn't get to fix this yet, so I'm glad about this.
> 
> I've added a couple of comments, will be great if you can take a look. Also worth verifying the the [integration tests](https://github.com/elementary-data/dbt-data-reliability/blob/master/CONTRIBUTING.md#running-integration-tests) pass, but will also check that.

For sure! This personally cuts down our Bigquery consumption by ~7% and is huge in cost savings. This matters a lot when the data volume is huge (TB, PBs)

dongchris commented:
> Thanks @dongchris for the awesome contribution! Actually realized as well just recently that `count(*)` does a full scan in BQ and didn't get to fix this yet, so I'm glad about this.
> 
> I've added a couple of comments, will be great if you can take a look. Also worth verifying the the [integration tests](https://github.com/elementary-data/dbt-data-reliability/blob/master/CONTRIBUTING.md#running-integration-tests) pass, but will also check that.

addressed your comments

was having trouble running the integration tests (followed the instructions on the elementary website) so if you could help check that, it would be great!
